app/assets/javascripts/discourse/app/components/emoji-picker/content.gjs

```diff
@@ -174,13 +179,14 @@ export default class EmojiPicker extends Component {
   debouncedDidInputFilter(filter = "") {
     filter = filter.toLowerCase();
 
-    this.filteredEmojis = this.flatEmojis.filter(
-      (emoji) =>
-        emoji.name.toLowerCase().includes(filter) ||
-        emoji.search_aliases?.any((alias) =>
-          alias.toLowerCase().includes(filter)
-        )
-    );
+    const results = emojiSearch(filter, {
+      exclude: this.site.denied_emojis,
+    }).slice(0, 25);
+
+    this.filteredEmojis =
+      this.flatEmojis.filter((emoji) => results.includes(emoji.name)) ?? [];
```

Comment:
Technically all of this is very wasteful, we iterate multiple times over the emoji list. We could use a better data structure, but this is way out of scope here.